### PR TITLE
Helm Charts for OpsSight

### PR DESCRIPTION
**Pull request by gautambaghel**

### CODE REVIEW

File: charts/opssight/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+*/}}
+{{- define "ops.selectorLabelsWithoutVersion" -}}
+app: opssight
+name: {{ .Release.Name }}
+{{- end -}}
+
+{{/*
+Security Context if Kubernetes
+*/}}
+{{- define "ops.podSecurityContext" -}}
```
Comment (msenmurugan) [inline charts/opssight/templates/_helpers.tpl:70]: remove it.. it won't be required

File: charts/opssight/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+imagePullSecrets:
+{{- toYaml . | nindent 0 }}
+{{- end }}
+{{- end }}
+
+{{/*
+Enable Binary Scanner
+*/}}
+{{- define "enableBinaryScanner" -}}
+{{- if .Values.enableBinaryScanner -}}
```
Comment (msenmurugan) [inline charts/opssight/templates/_helpers.tpl:91]: remove it

Comment (gautambaghel) [reply charts/opssight/templates/_helpers.tpl:91]: Thanks Senthil!

File: charts/opssight/values.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    port: 3001
+    timings:
+      checkForStalledScansPauseHours: 999999
+      stalledScanClientTimeoutHours: 999999
+      modelMetricsPauseSeconds: 15
+      unknownImagePauseMilliseconds: 15000
+      clientTimeoutMilliseconds: 100000
+    useMockMode: false
+    expose: "OpenShift" #[None|LoadBalancer|NodePort|OpenShift]
+  perceiver:
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:41]: for downstream, please rename all perceiver to processor

File: charts/opssight/values.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    enableImageProcessor: true
+    enablePodProcessor: true
+    enableQuayProcessor: true
+    exposeQuayProcessor: "OpenShift" #[None|LoadBalancer|NodePort|OpenShift]
+  scannerPod:
+    scanner:
+      port: 3003
+      imageDirectory: "/var/images"
+      blackDuckClientTimeoutSeconds: 600
+    imageFacade:
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:61]: rename image facade to image puller

File: charts/opssight/values.yaml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+# Specifications as it will go to configMap
+spec:
+  logLevel: "debug"
+  enableMetrics: true
+  exposeMetrics: "OpenShift" #[None|LoadBalancer|NodePort|OpenShift]
+  blackduck:
+    connectionsEnvironmentVariableName: "blackduck.json"
+    tlsVerification: false
+  perceptor:
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:31]: perceptor to core

File: charts/opssight/values.yaml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+# Secrets base64 encoded
+secret:
+  externalHub: e30=
+  securedRegistries: e30=
+
+# Specifications as it will go to configMap
+spec:
+  logLevel: "debug"
+  enableMetrics: true
+  exposeMetrics: "OpenShift" #[None|LoadBalancer|NodePort|OpenShift]
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:27]: make the default as None

File: charts/opssight/values.yaml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      clientTimeoutMilliseconds: 100000
+    useMockMode: false
+    expose: "OpenShift" #[None|LoadBalancer|NodePort|OpenShift]
+  perceiver:
+    port: 3002
+    certificate: ""
+    certificateKey: ""
+    annotationIntervalSeconds: 30
+    dumpIntervalMinutes: 30
+    podPerceiver:
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:47]: rename podPerceiver to podProcessor

File: charts/opssight/values.yaml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+init:
+  registry: # override the docker registry at container level
+  imageTag: 1.0.0
+  securityContext: {}
+
+prometheus:
+  registry: docker.io/prom # override the docker registry at container level
+  imageTag: v2.1.0
+  limitMemory: 640Mi
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:76]: resources changes similar to another container...

File: charts/opssight/templates/secrets.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+data:
+  blackduck.json: {{ .Values.secret.externalHub }}
+  securedRegistries.json: {{ .Values.secret.securedRegistries }}
+kind: Secret
+metadata:
+  creationTimestamp: null
```
Comment (msenmurugan) [inline charts/opssight/templates/secrets.yaml:7]: remove creationTimeStamp

File: charts/opssight/templates/secrets.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+data:
+  blackduck.json: {{ .Values.secret.externalHub }}
```
Comment (msenmurugan) [inline charts/opssight/templates/secrets.yaml:3]: add b64enc method like below. check for alert or blackduck helm chart. input should be not encoded
blackduck.json: {{ .Values.secret.externalHub | b64enc }}

Comment (msenmurugan) [reply charts/opssight/templates/secrets.yaml:3]: same changes to all secret values

File: charts/opssight/templates/opssight-pod-processor.yaml [hunk 11]
```diff
@@ -0,0 +1,144 @@
+apiVersion: apps/v1
```
Comment (msenmurugan) [inline charts/opssight/templates/opssight-pod-processor.yaml:1]: enable pod processor if block is missing

File: charts/opssight/templates/opssight-core.yaml [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+spec:
+  subdomain: ''
+  to:
+    kind: Service
+    name: {{ .Release.Name }}-opssight-core
+    weight: 100
+  port:
+    targetPort: port-core
+  wildcardPolicy: None
+{{ else }}
```
Comment (msenmurugan) [inline charts/opssight/templates/opssight-core.yaml:139]: remove else

File: charts/opssight/templates/opssight-artifactory-processor.yaml [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+spec:
+  subdomain: ''
+  to:
+    kind: Service
+    name: {{ .Release.Name }}-opssight-artifactory-processor
+    weight: 100
+  port:
+    targetPort: port-artifactory-processor
+  wildcardPolicy: None
+{{ else }}
```
Comment (msenmurugan) [inline charts/opssight/templates/opssight-artifactory-processor.yaml:145]: remove else

File: charts/opssight/values.yaml [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+# This is used to start or stop the app. Set to Running to start, or Stopped to stop.
+status: Running
+
+# Secrets base64 encoded
+secret:
+  externalHub: e30=
+  securedRegistries: e30=
+
+# Specifications as it will go to configMap
+spec:
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:24]: why it's having spec?

Comment (msenmurugan) [reply charts/opssight/values.yaml:24]: remove spec and move all the metrics related parameters under prometheus pod

Comment (gautambaghel) [reply charts/opssight/values.yaml:24]: Should I just remove Spec?

File: charts/opssight/values.yaml [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
+# image pull secret to download the Black Duck images (mostly applicable for air gapped customers)
+imagePullSecrets: [] # array of strings delimited by comma
+
+# This is used to start or stop the app. Set to Running to start, or Stopped to stop.
+status: Running
+
+# Secrets base64 encoded
+secret:
+  externalHub: e30=
```
Comment (msenmurugan) [inline charts/opssight/values.yaml:20]: have it plain text and in the secret do base64 encoded

File: charts/blackduck-connector/values.yaml [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+#   {
+#       "Url": "<EXTERNAL_REGISTRY_URL>",
+#       "user": "<EXTERNAL_REGISTRY_USERNAME>",
+#       "password": "<EXTERNAL_REGISTRY_PASSWORD>",
+#       "token": ""
+#   }
+# }
+
+secret:
+  externalHub: {}
```
Comment (msenmurugan) [inline charts/blackduck-connector/values.yaml:45]: pls change it to externalBlackDuck

File: charts/blackduck-connector/values.yaml [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+    requests:
+      cpu: 300m
+      memory: 1300Mi
+
+imageGetter:
+  registry:
+  imageTag:
+  port: 3004
+  host: "localhost"
+  imagePullerType: "docker"
```
Comment (msenmurugan) [inline charts/blackduck-connector/values.yaml:118]: Is the default imagePullerType is Docker in Operator? Can you pls double check, I thought it is skopeo.

File: charts/blackduck-connector/templates/secrets.yaml [hunk 18]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+data:
+  {{ .Values.blackduck.connectionsEnvironmentVariableName }}: {{ .Values.secret.externalHub | toJson | b64enc }}
+  securedRegistries.json: {{ .Values.secret.securedRegistries | toJson | b64enc }}
```
Comment (msenmurugan) [inline charts/blackduck-connector/templates/secrets.yaml:4]: you have tested with securedRegistries right??

File: charts/blackduck-connector/templates/opssight-scanner.yaml [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+        - containerPort: {{ .Values.imageGetter.port }}
+          protocol: TCP
+        resources:
+          {{- toYaml .Values.imageGetter.resources | nindent 12 }}
+        securityContext:
+          privileged: false
+        volumeMounts:
+        - mountPath: /etc/image-getter
+          name: image-getter
+        - mountPath: /var/images
```
Comment (msenmurugan) [inline charts/blackduck-connector/templates/opssight-scanner.yaml:95]: does the mount path is from .Values.scanner.imageDirectory?

File: charts/blackduck-connector/templates/opssight-scanner.yaml [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+        - containerPort: {{ .Values.scanner.port }}
+          protocol: TCP
+        resources:
+          {{- toYaml .Values.scanner.resources | nindent 12 }}
+        securityContext:
+          privileged: false
+        volumeMounts:
+        - mountPath: /etc/scanner
+          name: scanner
+        - mountPath: /var/images
```
Comment (msenmurugan) [inline charts/blackduck-connector/templates/opssight-scanner.yaml:62]: does the mount path is from .Values.scanner.imageDirectory?

File: charts/blackduck-connector/templates/secrets.yaml (hunk 18 above)
Comment (msenmurugan) [reply charts/blackduck-connector/templates/secrets.yaml:4]: and also external hub scenario?

File: charts/blackduck-connector/values.yaml [hunk 21]
```diff
@@ -0,0 +1,153 @@
...
+# image pull secret to download the Black Duck images (mostly applicable for air gapped customers)
+imagePullSecrets: [] # array of strings delimited by comma
+
+# This is used to start or stop the app. Set to Running to start, or Stopped to stop.
+status: Running
+
+logLevel: "debug"
+
+# External Hub ~>
+#  {
```
Comment (msenmurugan) [inline charts/blackduck-connector/values.yaml:21]: These examples are json and also an external Black Duck is an array

File: charts/blackduck-connector/values.yaml (hunk 17 above)
Comment (gautambaghel) [reply charts/blackduck-connector/values.yaml:118]: I think you're right, it's skopeo

File: charts/blackduck-connector/values.yaml [hunk 22]
```diff
@@ -0,0 +1,153 @@
...
+#         "domain":"<BLACK_DUCK_URL>",
+#         "port":443,
+#         "user":"<BLACK_DUCK_USERNAME>",
+#         "password":"<BLACK_DUCK_PASSWORD>",
+#         "concurrentScanLimit":3
+#     }
+#   }
+
+# Secured Registries ~> 
+# {
```
Comment (msenmurugan) [inline charts/blackduck-connector/values.yaml:34]: Similarly secured registries is an array

File: charts/blackduck-connector/templates/secrets.yaml (hunk 18 above)
Comment (gautambaghel) [reply charts/blackduck-connector/templates/secrets.yaml:4]: Both tested :) Senthil, my artifactory and quay processors won't work without it :)

File: charts/blackduck-connector/values.yaml (hunk 21 above)
Comment (gautambaghel) [reply charts/blackduck-connector/values.yaml:21]: I think operator kinda did that conversion from array to map internally, cos I tried array and it won't work. It needs a map

File: charts/blackduck-connector/values.yaml (hunk 22 above)
Comment (gautambaghel) [reply charts/blackduck-connector/values.yaml:34]: I think operator kinda did that conversion from array to map internally, cos I tried array and it won't work. It needs a map

File: charts/blackduck-connector/templates/opssight-scanner.yaml (hunk 19 above)
Comment (gautambaghel) [reply charts/blackduck-connector/templates/opssight-scanner.yaml:95]: done

File: charts/blackduck-connector/templates/opssight-scanner.yaml (hunk 20 above)
Comment (gautambaghel) [reply charts/blackduck-connector/templates/opssight-scanner.yaml:62]: done

File: charts/blackduck-connector/values.yaml (hunk 16 above)
Comment (gautambaghel) [reply charts/blackduck-connector/values.yaml:45]: done

File: charts/blackduck-connector/values.yaml (hunk 21 above)
Comment (msenmurugan) [reply charts/blackduck-connector/values.yaml:21]: pls provide an example in yaml format

**Pull request merged**